src/common/k8s-api/api-manager/api-manager.ts

```diff
@@ -33,20 +33,38 @@ export class ApiManager {
   private readonly externalApis = observable.array<KubeApi>();
   private readonly externalStores = observable.array<KubeObjectStore>();
 
-  constructor(private readonly dependencies: Dependencies) {}
+  private readonly apis = observable.map<string, KubeApi>();
+
+  constructor(private readonly dependencies: Dependencies) {
+    // NOTE: this is done to preserve the old behaviour of an API being discoverable using all previous apiBases
+    autorun(() => {
```

Comment:
No, because the fact that it is stateful is the point of being used. To maintain both backwards comparability and the expected behaviour we want to maintain a list of all the `apiBase`'s that were ever used even if they are currently different.